### PR TITLE
Add breadcrumbs to help app

### PR DESCRIPTION
**Pull request by msmanoj**
## Type

- [ ] Bug fix
- [x] New feature
- [ ] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1385

## Description
- Created a shared breadcrumbs component
- Added breadcrumbs to help app

## Deployment URL
http://help-app-breadcrumbs.review.ensembl.org

## Views affected
- Help app

### Other effect
- [ ] I have checked any requirements for Google Analytics
- [x] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

### CODE REVIEW

File: src/ensembl/src/content/app/help/Help.tsx [hunk 1]
```diff
@@ -90,18 +107,69 @@ const MainContent = (props: { article: ArticleData }) => {
...
+    iteration 3 gets `items.1.items.0.items.1.name` from menu
+  */
+  for (let i = 1; i <= pathPortions?.length / 2; i++) {
+    const slicedPath = [...pathPortions.slice(0, i * 2), 'name'];
+
+    breadcrumbs.push(get(menu, slicedPath));
+  }
+
+  return breadcrumbs;
+};
```
[the remark this answers is not in the document]
Comment (jyothishnt) [reply src/ensembl/src/content/app/help/Help.tsx:176]: I also go with @azangru, it is much cleaner. Thanks

[the remark this answers is not in the document]
Comment (imransl) [reply src/ensembl/src/content/app/help/Help.tsx:176]: Yes let's go with this

[the remark this answers is not in the document]
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.tsx:160]: I've updated it 👍  Thanks

File: src/ensembl/src/content/app/help/Help.scss [hunk 2]
```diff
@@ -23,10 +23,24 @@
 
 .main {
   height: 100%;
-  padding-left: 80px;
-  padding-top: 44px;
+  display: grid;
+  padding-left: 44px;
+  padding-top: 25px;
+  grid-template-areas: "breadcrumbs breadcrumbs navigation"
+                       "space  article  article";
```
Comment (azangru) [inline src/ensembl/src/content/app/help/Help.scss:30]: If you want to skip a column, the css-grid convention is to use a dot ([docs](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Grid_Template_Areas#leaving_a_grid_cell_empty))

File: src/ensembl/src/content/app/help/Help.tsx [hunk 3]
```diff
@@ -49,12 +53,23 @@ const Help = () => {
...
+      <div className={styles.breadcrumbsContainer}>
+        <Breadcrumbs breadcrumbs={['Overview']} />
+      </div>
+      <div className={styles.articleContainer}>
+        <HelpLanding />
+      </div>
     </main>
-  ) : article ? (
-    <MainContent article={article} />
+  ) : article && breadcrumbs ? (
```
Comment (azangru) [inline src/ensembl/src/content/app/help/Help.tsx:71]: `&& breadcrumbs` is always true if it's initialised with an empty array.

File: src/ensembl/src/content/app/help/Help.tsx [hunk 4]
```diff
@@ -49,12 +53,23 @@ const Help = () => {
     endpoint: `/api/docs/article?url=${encodeURIComponent(location.pathname)}`,
     skip: isIndexPage
   });
+
+  let breadcrumbs: string[] | undefined = [];
+  if (!isIndexPage && menu && article) {
```
Comment (azangru) [inline src/ensembl/src/content/app/help/Help.tsx:58]: Why do we exclude the index page from the search, by the way? Won't `buildBreadcrumbs` find the "Overview" item? I would have thought that it should.

File: src/ensembl/src/shared/components/breadcrumbs/Breadcrumbs.scss [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+.breadcrumbs {
+  color: $black;
+  font-weight: $light;
+
+  .breadcrumb{
+    margin-right: 15px;
+  }
+
+  .breadcrumb::after {
```
Comment (azangru) [inline src/ensembl/src/shared/components/breadcrumbs/Breadcrumbs.scss:11]: `.breadcrumb:not(:last-child)::after`? Then there will be no reason to reset the content on line 20.

File: src/ensembl/src/shared/components/breadcrumbs/Breadcrumbs.test.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    const { container } = render(<Breadcrumbs {...defaultProps} />);
+    expect(container.querySelector('.breadcrumbs')).toBeTruthy();
+  });
+
+  it('renders the correct number breadcrumbs', () => {
+    const { container } = render(<Breadcrumbs {...defaultProps} />);
+    expect(container.querySelectorAll('.breadcrumb')).toHaveLength(
+      defaultProps.breadcrumbs.length
+    );
+  });
```
Comment (azangru) [inline src/ensembl/src/shared/components/breadcrumbs/Breadcrumbs.test.tsx:41]: This test can probably be removed; the next test does the same check and more.

File: src/ensembl/src/content/app/help/Help.scss (hunk 2 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.scss]: That's done

File: src/ensembl/src/content/app/help/Help.tsx [hunk 7]
```diff
@@ -49,12 +53,23 @@ const Help = () => {
     endpoint: `/api/docs/article?url=${encodeURIComponent(location.pathname)}`,
     skip: isIndexPage
   });
+
+  let breadcrumbs: string[] | undefined = [];
```
[the remark this answers is not in the document]
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.tsx]: I've removed `undefined` from L57 and returned empty [] from `buildBreadcrumbs` by default to keep it simple.

File: src/ensembl/src/content/app/help/Help.tsx (hunk 3 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.tsx:71]: Please refer to the comment above

File: src/ensembl/src/content/app/help/Help.tsx (hunk 4 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.tsx]: That's right.  I've updated it.

File: src/ensembl/src/shared/components/breadcrumbs/Breadcrumbs.scss (hunk 5 above)
Comment (msmanoj) [reply src/ensembl/src/shared/components/breadcrumbs/Breadcrumbs.scss]: That's done

File: src/ensembl/src/shared/components/breadcrumbs/Breadcrumbs.test.tsx (hunk 6 above)
Comment (msmanoj) [reply src/ensembl/src/shared/components/breadcrumbs/Breadcrumbs.test.tsx]: That's done

File: src/ensembl/src/content/app/help/Help.test.tsx [hunk 8]
```diff
@@ -81,8 +81,8 @@ describe('buildBreadcrumbs', () => {
     ]);
   });
 
-  it('returns undefined when there is no match', () => {
-    expect(buildBreadcrumbs(sampleMenu, { url: '/blah' })).toEqual(undefined);
+  it('returns [] when there is no match', () => {
```
Comment (azangru) [inline src/ensembl/src/content/app/help/Help.test.tsx:84]: `[]` -> `an empty array`?

File: src/ensembl/src/content/app/help/Help.tsx (hunk 3 above)
Comment (azangru) [reply src/ensembl/src/content/app/help/Help.tsx:71]: I'm confused. Now `breadcrumbs` is definitely always truthy. Shouldn't you remove them from the check?

File: src/ensembl/src/content/app/help/Help.tsx [hunk 9]
```diff
@@ -28,8 +28,12 @@ import {
   HelpArticleGrid,
   VideoArticle
 } from 'src/shared/components/help-article';
+import Breadcrumbs from 'ensemblRoot/src/shared/components/breadcrumbs/Breadcrumbs';
```
Comment (jyothishnt) [inline src/ensembl/src/content/app/help/Help.tsx:31]: ensemblRoot

File: src/ensembl/src/content/app/help/Help.test.tsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+          type: 'article',
+          url: '/six'
+        },
+        {
+          name: 'Seven',
+          type: 'article',
+          url: '/seven'
+        },
+        {
+          name: 'Two',
```
Comment (jyothishnt) [inline src/ensembl/src/content/app/help/Help.test.tsx:59]: I know it doesnt matter much but did you mean Eight here?

**Comment by jyothishnt** [comment]
Minor - the font size of breadcrumbs looks much smaller than what I see in the review app
Screenshot 2021-11-04 at 15 46 39

File: src/ensembl/src/content/app/help/Help.test.tsx (hunk 8 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.test.tsx:84]: That's done

File: src/ensembl/src/content/app/help/Help.tsx (hunk 3 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.tsx:71]: I've removed it 👍

File: src/ensembl/src/content/app/help/Help.tsx (hunk 9 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.tsx:31]: Removed it

File: src/ensembl/src/content/app/help/Help.test.tsx (hunk 10 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.test.tsx:59]: Changed it 😄

**Comment by msmanoj** [comment]
> Minor - the font size of breadcrumbs looks much smaller than what I see in the review app
> 
> Screenshot 2021-11-04 at 21 38 46

I think it is 1px smaller. I've changed it.

**Review by azangru — APPROVED** [review]

**Pull request merged**